llvm/include/llvm/CodeGen/BasicTTIImpl.h

```diff
@@ -565,7 +565,12 @@ class BasicTTIImplBase : public TargetTransformInfoImplCRTPBase<T> {
       if (!isa<ConstantInt>(Inst.getOperand(1)))
         return false;
       EVT VT = getTLI()->getValueType(DL, Inst.getType());
-      return !getTLI()->isIntDivCheap(VT, Fn.getAttributes());
+
+      bool IsSigned = true;
+      if (Inst.getOpcode() == Instruction::SDiv ||
+          Inst.getOpcode() == Instruction::SRem)
+        IsSigned = false;
```

Comment:
can just initialize isSigned with the compare result

Reply:
I don't think we need to differentiate signed vs unsigned here.
Both kinds of division are expensive. This transformation is necessary in all cases.
That should avoid adding isSigned everywhere.